libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs

```diff
@@ -207,9 +211,19 @@ queryCommitteeMembersState coldCredsFilter hotCredsFilter statusFilter nes = do
                 Nothing -> MemberNotAuthorized
                 Just Nothing -> MemberResigned
                 Just (Just hk) -> MemberAuthorized hk
-        -- TODO: implement after pulsing and snapshots are done
-        let nextEpochChange = NoChangeExpected
-        pure $ CommitteeMemberState hkStatus status mbExpiry nextEpochChange
+        pure $ CommitteeMemberState hkStatus status mbExpiry (nextEpochChange coldCred)
+
+      nextEpochChange :: Credential 'ColdCommitteeRole (EraCrypto era) -> NextEpochChange
+      nextEpochChange ck
+        | inCurrent && not inNext = ToBeRemoved
+        | not inCurrent && inNext = ToBeEnacted
+        | otherwise = NoChangeExpected
+        where
+          -- This will also include members that are in none of the Committes,
+          -- but only in CommitteeState (Unrecognized members)
```

Comment:
I think this comment is very confusing. Below bindings only check membership in current and in future committee, which is clear from the code, extra explanation will only add confusion
```suggestion
```